gfx/common/x11_common.c

```diff
@@ -306,9 +306,17 @@ static void x11_suspend_screensaver_xdg_screensaver(Window wnd, bool enable)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
-    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
 #ifdef HAVE_DBUS
     x11_suspend_screensaver_dbus(enable);
+    if (dbus_screensaver_cookie == 0)
+    {
+        // Only attempt xdg-screensaver method if dbus didn't succeed
+#endif
+
+        x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+
+#ifdef HAVE_DBUS
+    }
```

Comment:
> Sorry I don't understand this, could you expand? Do you mean there's a bug in the DBus suspension?

dbus uninhibiting is unimplemented (not sure about xdg), which counts as a bug. In the current codebase, this bug/hole is currently bounded to that function (nothing else needs to be changed to fix it); I want it to remain that way.

...actually, unsuspending can't depend on post-call state at all, it's gonna be zero after trying to dbus-uninhibit whether it was originally suspended or not. We have to either (A) check if it was dbus-inhibited BEFORE the uninhibited call, (B) put that in the return value, or (C) have uninhibit always poke both (but keep the check if suspending), removing a nonexistent inhibition isn't going to do anything.

(A) is ugly, but I have no real preference between (B) or (C).

> but then realised we already has the cookie variable telling us if it worked or not and that's already used in all the other DBus methods

But this is not a DBus method. It's the generic 'suspend X11 screensaver by any means necessary' function. All functions that poke that cookie have 'dbus' in the name, and I'd prefer it if remains that way.

If you don't want to return a value from inhibit-screensaver-via-dbus(), a is-screensaver-inhibited-via-dbus() function that just returns that cookie would also be a valid implementation.